src/vcs/git.rs

```diff
@@ -171,6 +184,30 @@ impl<'a> Browser<'a> {
         })
     }
 
+    /// Switch the namespace you are browsing in. This will consume the previous
+    /// `Browser` and give you back a new `Browser` for that particular
+    /// namespace. The `revision` provided will kick-off the history for
+    /// this `Browser`.
+    pub fn switch_namespace(self, namespace: &Namespace, revision: &str) -> Result<Self, Error> {
+        self.repository.switch_namespace(&namespace.to_string())?;
+        let history = self.repository.get_history(revision.to_string())?;
+        Ok(Browser {
+            snapshot: self.snapshot,
+            repository: self.repository,
+            history,
+        })
+    }
```

Comment:
Doesn't it make more sense to have a `Browser::new_with_namespace` instead?

Reply:
I don't think a `new` version would make sense because we would take `self` to change the repository's namespace. It could be debated, however, whether this _needs_ to consume the `Browser` and give you back a new one or rather just take the reference and switch the namespace.

Reply:
I'm not saying we should rename the function, but that it's more useful to create a new browser from a namespace, than to create a new browser and then create a new browser from an old browser just to switch the namespace!

Reply:
Ach so! Ya perfect idea :)